sdk/core/azure-core-tracing-opentelemetry/src/samples/AsyncListKeyVaultSecretsSample.md

```diff
@@ -0,0 +1,107 @@
+### Adding dependencies to your project:
+```xml
+<dependencies>
+    <dependency>
+        <groupId>io.opentelemetry</groupId>
+        <artifactId>opentelemetry-sdk</artifactId>
+        <version>0.2.0-SNAPSHOT</version>
+    </dependency>
+    <dependency>
+        <groupId>com.azure</groupId>
+        <artifactId>azure-security-keyvault-secrets</artifactId>
+        <version>4.0.0</version>
+    </dependency>
+    <dependency>
+        <groupId>com.azure</groupId>
+        <artifactId>azure-core-tracing-opentelemetry</artifactId>
+        <version>1.0.0-beta.1</version>
+    </dependency>
+    <dependency>
+        <groupId>io.opentelemetry</groupId>
+        <artifactId>opentelemetry-exporters-logging</artifactId>
+        <version>0.2.0-SNAPSHOT</version>
+    </dependency>
+</dependencies>
+```
+
+#### Sample demonstrates tracing when asynchronously creating and listing secrets from a Key Vault using [azure-security-keyvault-secrets][azure_keyvault_secrets] client library.
+```java
+import com.azure.identity.DefaultAzureCredentialBuilder;
+import com.azure.security.keyvault.secrets.SecretAsyncClient;
+import com.azure.security.keyvault.secrets.SecretClientBuilder;
+import com.azure.security.keyvault.secrets.models.KeyVaultSecret;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.sdk.trace.TracerSdkFactory;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.Tracer;
+import reactor.util.context.Context;
+
+import java.util.logging.Logger;
+
+import static com.azure.core.util.tracing.Tracer.PARENT_SPAN_KEY;
+import static java.util.logging.Logger.getLogger;
+
+/**
+ * Sample demonstrates tracing how to add and list secrets in a Key vault with tracing enabled with a Logging Exporter.
+ */
+public class Sample {
+
+    private static final Logger LOGGER = getLogger("Sample");
+    private static final Tracer TRACER;
+    private static final TracerSdkFactory TRACER_SDK_FACTORY;
+
+    static {
+        TRACER_SDK_FACTORY = Helper.configureOpenTelemetryAndJaegerExporter(LOGGER);
+        TRACER = TRACER_SDK_FACTORY.get("Azure-OpenTelemetry");
+    }
+
+    public static void main(String[] args) throws InterruptedException {
+        Span userSpan = TRACER.spanBuilder("user-parent-span").startSpan();
+        final Scope scope = TRACER.withSpan(userSpan);
+        LOGGER.info("=== Start user scoped span  ===");
+        doClientWork();
+        LOGGER.info("=== End user scoped span  ===");
+        userSpan.end();
+        scope.close();
+        TRACER_SDK_FACTORY.shutdown();
+        LOGGER.info("=== Tracer Shutdown  ===");
+    }
+    
+    public static void doClientWork() throws InterruptedException {
+        SecretAsyncClient client = new SecretClientBuilder()
+            .vaultUrl("YOUR_VAULT_URL")
+            .credential(new DefaultAzureCredentialBuilder().build())
+            .buildAsyncClient();
+
+        Context traceContext = Context.of(PARENT_SPAN_KEY, TRACER.getCurrentSpan());
```

Comment:
BTW, is it needed if user only calls sync APIs? Policies should fallback to the current span when there is nothing in the context. 

Can we articulate it somehow with examples when it's needed vs not needed?

Reply:
Added a note [here][https://github.com/Azure/azure-sdk-for-java/pull/6131/files#diff-040661d2aa5706ffc7cb8e3b9daa1db2R85] which mentions fallback option when no current span passed by the user.
Also added an [example](https://github.com/Azure/azure-sdk-for-java/pull/6131/files#diff-3f98e37f1a0f96922dadd5266ebaa062R82) showing the same.